File: src/main/java/no/nav/bidrag/beregn/underholdskostnad/UnderholdskostnadCore.java
package no.nav.bidrag.beregn.underholdskostnad;

import no.nav.bidrag.beregn.underholdskostnad.dto.BeregnUnderholdskostnadGrunnlagCore;
import no.nav.bidrag.beregn.underholdskostnad.dto.BeregnetUnderholdskostnadResultatCore;
import no.nav.bidrag.beregn.underholdskostnad.periode.UnderholdskostnadPeriode;

public interface UnderholdskostnadCore {

  BeregnetUnderholdskostnadResultatCore beregnUnderholdskostnad(BeregnUnderholdskostnadGrunnlagCore beregnUnderholdskostnadGrunnlagCore);

  static UnderholdskostnadCore getInstance() {
    return new UnderholdskostnadCoreImpl(UnderholdskostnadPeriode.Companion.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/underholdskostnad/UnderholdskostnadCore.kt
package no.nav.bidrag.beregn.underholdskostnad

import no.nav.bidrag.beregn.underholdskostnad.dto.BeregnUnderholdskostnadGrunnlagCore
import no.nav.bidrag.beregn.underholdskostnad.dto.BeregnetUnderholdskostnadResultatCore
import no.nav.bidrag.beregn.underholdskostnad.periode.UnderholdskostnadPeriode

fun interface UnderholdskostnadCore {
    fun beregnUnderholdskostnad(grunnlag: BeregnUnderholdskostnadGrunnlagCore): BeregnetUnderholdskostnadResultatCore

    companion object {
        fun getInstance(): UnderholdskostnadCore = UnderholdskostnadCoreImpl(UnderholdskostnadPeriode.getInstance())
    }
}
